### PR TITLE
Remove deprecated positional arguments to map methods

### PR DESCRIPTION
**Pull request by dstansby**
These were deprecated in 2.0 (an LTS), so good to go in 2.1.

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

**Comment by ayshih** [comment]
It's not an "obsolete" decorator.  The decorator rewrites the function signature so that the arguments are not positional only.  See #3677 

This was deprecated in 2.0, so it's okay to remove (I think), but the motivation here is not correct.

**Review by ayshih — CHANGES_REQUESTED** [review]
I'll block this until there's a discussion about whether we want to remove this deprecated behavior.

**Comment by nabobalis** [comment]
> It's not an "obsolete" decorator. The decorator rewrites the function signature so that the arguments are not keyword only. See #3677
> 
> This was deprecated in 2.0, so it's okay to remove (I think), but the motivation here is not correct.

Is it not obsolete in the sense that we don't need it anymore? I still forgot if we remove this in 2.1 or 3.0+ but as far as I can see it seems to be ok for me. @Cadair can be the final judge.

**Comment by Cadair** [comment]
By the deprecation policy this is up for removal, so pulling it seems fine to me unless people think we should move slower in methods of Map?

**Comment by ayshih** [comment]
Sure, but @dstansby appears to mean "obsolete" in the sense that he thought that the decorator wasn't do anything anymore, but it most definitely is.  That is, the decorator makes it so the function can accept positional arguments (and raises a deprecation warning), even though the function signature looks like it can't.  The PR should state a valid reason for a change, even if the change is acceptable by some other policy.

**Comment by dstansby** [comment]
Ah I got confused and thought that the signature as written on the decorated function was the signature, and the decorator just raised a warning if arguments were passed as positional arguments (this is how our decorators work in Matplotlib). Seems like these are fine to remove with appropriate updates to the tests and a changelog though so I'll update this PR as such.

**Comment by ayshih** [comment]
> By the deprecation policy this is up for removal, so pulling it seems fine to me unless people think we should move slower in methods of Map?

Maybe it's just how COVID has affected my sense of time, but it certainly doesn't feel like it's been half a year since this functionality was deprecated.  <shrug>

**Review by ayshih — COMMENTED** [review]
.

**Comment by nabobalis** [comment]
> Sure, but @dstansby appears to mean "obsolete" in the sense that he thought that the decorator wasn't do anything anymore, but it most definitely is. That is, the decorator makes it so the function can accept positional arguments (and raises a deprecation warning), even though the function signature looks like it can't. The PR should state a valid reason for a change, even if the change is acceptable by some other policy.

Good point we should remove the * in each function.

**Comment by dstansby** [comment]
I think it's been 6 months (since 2.0), but since the deprecation is in an LTS release it's good to remove now (https://github.com/sunpy/sunpy-SEP/blob/master/SEP-0009.md)

**Comment by dstansby** [comment]
> > Sure, but @dstansby appears to mean "obsolete" in the sense that he thought that the decorator wasn't do anything anymore, but it most definitely is. That is, the decorator makes it so the function can accept positional arguments (and raises a deprecation warning), even though the function signature looks like it can't. The PR should state a valid reason for a change, even if the change is acceptable by some other policy.
> 
> Good point we should remove the * in each function.

No we shouldn't, that's what makes the arguments keyword only!

**Comment by nabobalis** [comment]
> > > Sure, but @dstansby appears to mean "obsolete" in the sense that he thought that the decorator wasn't do anything anymore, but it most definitely is. That is, the decorator makes it so the function can accept positional arguments (and raises a deprecation warning), even though the function signature looks like it can't. The PR should state a valid reason for a change, even if the change is acceptable by some other policy.
> > 
> > 
> > Good point we should remove the * in each function.
> 
> No we shouldn't, that's what makes the arguments keyword only!

Remove the other one then.

**Review by ayshih — APPROVED** [review]
I don't know what's going on with our test suite, but the code changes look good

File: changelog/4616.breaking.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+The following `~sunpy.map.Map` methods have had support for specific positional
+arguments removed. They must now be passed as keyword arguments
+(i.e. ``m.method(keyword_arg=value)``).
+
+- :meth:`~sunpy.map.Map.submap`: ``width``, ``height``.
+- :meth:`~sunpy.map.Map.draw_rectangle`: ``width``, ``height``, ``axes``, ``top_right``.
```
Comment (Cadair) [inline changelog/4616.breaking.rst:6]: does our changelog rendering actually respect whitespace like this?

**Review by Cadair — APPROVED** [review]
Looks good tho the style check is angry

File: changelog/4616.breaking.rst (hunk 1 above)
Comment (nabobalis) [reply changelog/4616.breaking.rst:6]: Only one way to find out.

Comment (Cadair) [reply changelog/4616.breaking.rst:6]: the line break isn't rendered but the list is.

**Pull request merged**